erpnext/stock/get_item_details.py

```diff
@@ -1171,14 +1175,28 @@ def get_projected_qty(item_code, warehouse):
 
 
 @frappe.whitelist()
-def get_bin_details(item_code, warehouse, company=None):
-	bin_details = frappe.db.get_value(
-		"Bin",
-		{"item_code": item_code, "warehouse": warehouse},
-		["projected_qty", "actual_qty", "reserved_qty"],
-		as_dict=True,
-		cache=True,
-	) or {"projected_qty": 0, "actual_qty": 0, "reserved_qty": 0}
+def get_bin_details(item_code, warehouse, company=None, include_child_warehouses=False):
+	bin_details = {"projected_qty": 0, "actual_qty": 0, "reserved_qty": 0}
+
+	if warehouse:
+		from frappe.query_builder.functions import Sum
+
+		from erpnext.stock.doctype.warehouse.warehouse import get_child_warehouses
+
+		warehouses = get_child_warehouses(warehouse) if include_child_warehouses else [warehouse]
+
+		bin = frappe.qb.DocType("Bin")
+		bin_details = (
+			frappe.qb.from_(bin)
+			.select(
+				Sum(bin.projected_qty).as_("projected_qty"),
+				Sum(bin.actual_qty).as_("actual_qty"),
+				Sum(bin.reserved_qty).as_("reserved_qty"),
+			)
+			.where((bin.item_code == item_code) & (bin.warehouse.isin(warehouses)))
+		).run(as_dict=True)[0]
+		bin_details = {k: 0 if not v else v for k, v in bin_details.items()}
```

Comment:
What is the use of this? bin_details is already a dictionary

Reply:
This is to replace the `None` values with `0`. This happens when there is no Bin for the key (item_code, warehouse). 

Example: 

![image](https://user-images.githubusercontent.com/63660334/209920729-c9fcd1dd-b2ce-47e5-bd14-bc0c7217850e.png)

![image](https://user-images.githubusercontent.com/63660334/209920758-8006a8be-1389-4b9c-a04c-3f4a0c9cd2f0.png)

Reply:
Let's use `frappe.db.get_value()` instead.

Reply:
Use COALESCE(SUM(projected_qty),0)